File: core/src/com/bombbird/terminalcontrol/entities/weather/VisibilityChance.java
package com.bombbird.terminalcontrol.entities.weather;

import com.badlogic.gdx.math.MathUtils;

public class VisibilityChance {
    public static int getRandomVis() {
        int randomNo = MathUtils.random(544);
        int counter = 0;
        for (int i = 1; i <= 9; i++) {
            if (randomNo <= counter) return i * 1000;
            counter += (i + 1);
        }
        return 10000;
    }
}

File: core/src/com/bombbird/terminalcontrol/entities/weather/VisibilityChance.kt
package com.bombbird.terminalcontrol.entities.weather

import com.badlogic.gdx.math.MathUtils

object VisibilityChance {
    val randomVis: Int
        get() {
            val randomNo = MathUtils.random(544)
            var counter = 0
            for (i in 1..9) {
                if (randomNo <= counter) return i * 1000
                counter += i + 1
            }
            return 10000
        }
}